### PR TITLE
POL-270 Filter for enabled column is always BOOLEAN_IS

### PR DESCRIPTION
**Pull request by pilhuhn**
Signed-off-by: Heiko W. Rupp <hrupp@redhat.com>

### CODE REVIEW

**Comment by pilhuhn** [comment]
@josejulio I wonder if we should remove the BOOLEAN_IS from the other columns.
And perhaps rewrite the entire Parameter list to only list filter[] and filter:op[] once, but with explanation and examples.

**Comment by josejulio** [comment]
> @josejulio I wonder if we should remove the BOOLEAN_IS from the other columns.

This makes sense, no point in using that `BOOLEAN_IS` in other columns.

> And perhaps rewrite the entire Parameter list to only list filter[] and filter:op[] once, but with explanation and examples.

It would work, but is more explicit separating the columns and actually giving the specific parameters that are valid for each one instead of all-to-all.

File: src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java [hunk 1]
```diff
@@ -118,10 +118,22 @@ public static Pager extractPager(@NotNull UriInfo uriInfo) {
...
                 String value = queryParams.getFirst(key);
-                String operatorString = queryParams.getFirst(String.format("%s[%s]", FILTER_OP, column));
-                Filter.Operator operator = Filter.Operator.EQUAL;
-                if (operatorString != null) {
-                    operator = Filter.Operator.fromName(operatorString);
+                Filter.Operator operator;
+                if (column.equals("is_enabled")) {
+                    operator = Filter.Operator.BOOLEAN_IS;
+                    if (value==null || value.isEmpty()) {
+                        value = "true";
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java:125]: +1

We could document that this value defaults to true when not found or empty:` @Parameter(name= filter[is_enabled]`

File: src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java [hunk 2]
```diff
@@ -118,10 +118,22 @@ public static Pager extractPager(@NotNull UriInfo uriInfo) {
             if (filterMatcher.find()) {
                 String column = filterMatcher.group(1);
                 String value = queryParams.getFirst(key);
-                String operatorString = queryParams.getFirst(String.format("%s[%s]", FILTER_OP, column));
-                Filter.Operator operator = Filter.Operator.EQUAL;
-                if (operatorString != null) {
-                    operator = Filter.Operator.fromName(operatorString);
+                Filter.Operator operator;
+                if (column.equals("is_enabled")) {
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java:122]: The only "problem" is that `PagingUtils` was meant to be general across all our resources. 
I think we will figure out what to do IF we want to have more resources filtered, not an issue right now.

Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java:122]: Yes. I had the same thoughts. We would need some datastructure (additional fields on the `Filter.Operator` enum) to list applicable fields or vice versa.
But yes, let's think about it when we need it.

File: src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java (hunk 1 above)
Comment (pilhuhn) [reply src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java:125]: I wonder if the @Param for filter:op[is_enabled] should just go away

Comment (josejulio) [reply src/main/java/com/redhat/cloud/policies/app/rest/utils/PagingUtils.java:125]: Ah, for this case it makes sense to just remove it, is the only possible value.

**Pull request merged**